AlphaWallet/alpha-wallet-ios
# Added Export JSON Keystore screens

Closes #2719

# Testing

## Preparation
1. Tap "Settings", "Advanced".
2. **Expects** to see "Advanced" screen with "Export JSON Keystore" option.

## Test Password screen
3. Tap "Export JSON Keystore".
4. **Expects** to see "Set Password for JSON Keystore" screen, with the password text field as first responder, the "Export JSON Keystore" button is disabled.

### Test Enable/Disable "Export JSON Keystore" button
5. Type "abc" into the password text field.
6. **Expects** "Export JSON Keystore" button to be enabled, password text field does not show password.
7. Tap the backspace key to delete a character.
8. **Expects** "Export JSON Keystore" button to be disabled.

### Test character filter (right now only alphanumerics are accepted)
9. Type "+++" into the password text field.
10. **Expects** nothing to happen.

### Test show/hide password
11. Tap the "Eye" button.
12. **Expects** the password to be shown in the password text field.
13. Type "1234567890" into the password text field.
14. **Expects** to see "ab1234567890" in the password field.

## Test Info screen
15. Tap "Export JSON Keystore" button.
16. **Expects** "Export JSON Keystore Info screen" with some random text displayed.

### Test back button
17. Tap the back button.
18. **Expects** to see the "Set Password for JSON Keystore" screen.

### Test complete export
19. Tap "Export JSON Keystore" button.
20. Tap the "Export Button".
21. **Expects** to return to the "Advanced" screen.

# Start Mode
In the `ExportJsonKeystoreCoordinator` class, you can specify which screen to start first via the `start` function.

# Validation rules
The validation rules are a list of rules which can specify the validity of the password. The rules are:
1. lengthLessThan
2. lengthLessThanOrEqualTo
3. lengthMoreThan
4. lengthMoreThanOrEqualTo
5. doesNotContain
6. onlyContain
Rules 1-4 expect an integer. The code does not check for logic. Rules 5-6 expect `CharacterSet` data.

# Test show export page 
## If a Watched Wallet
### Preparation
You will need an Etherium wallet address. You can get one from [Etherscan](https://etherscan.io/). The one I choose was `0xCD458d7F11023556cC9058F729831a038Cb8Df9c`.

1. Go to the main screen and tap "Wallet".
2. Create a watch only wallet by tapping on the upper left icon in the "Wallet" screen.
3. **Expects** to see "Wallets" screen with list of wallets.
4. Tap the add button in the upper right hand corner and select Watch Wallet.
5.  Enter the wallet address.
6. **Expects** to see the "Wallet" screen.
7. Tap on settings and advanced.
8. **Expects** not to see the Export functions.

## If a Real Wallet

1. Go to the main screen and tap "Wallet".
2. Create a wallet by tapping on the upper left icon in the "Wallet" screen.
3. **Expects to see "Wallets" screen with list of wallets.
4. Tap the add button in the upper right hand corner and select Create/Import.
5. Type in the key phase if you are importing a wallet.
6. **Expects** to see the "Wallet" screen.
7. Tap on settings and advanced.
8. **Expects** to see the Export functions.
